va_explorer/va_analytics/dash_apps/va_dashboard.py

```diff
@@ -712,66 +739,95 @@ def filter_data(
         # if no valid or invalid data, turn off timeframe
         if (len(valid_filter["ids"]) == 0) and (len(invalid_filter["ids"]) == 0):
             disable_timeframe = True
+        
 
         return json.dumps(combined_filter_dict), disable_timeframe
 
 
 # helper method to get filter ids given adjacent plot regions
 def _get_filter_dict(
     va_df,
-    selected_json,
+    selected_json={},
     timeframe="all",
     search_terms=[],
     locations=None,
     location_types=None,
+    restrictions=[]
 ):
 
     filter_df = va_df.copy()
     granularity = INITIAL_GRANULARITY
     plot_ids, plot_regions = list(), list()
-
+    
     filter_dict = {
-        "geo_filter": (selected_json is not None) or (len(search_terms) > 0),
+
+        "geo_filter": any(map(lambda x: len(x) > 0, [restrictions, selected_json, search_terms])),
+        "search_filter": (len(search_terms) > 0),
         "plot_regions": [],
         "chosen_region": "all",
         "ids": [],
         "plot_ids": [],
     }
+    
     if filter_dict["geo_filter"]:
-        # first, check if user searched anything. If yes, use that as filter.
+        # first, check if geo filter is for location restrictions
+        if len(restrictions) > 0:
+            # TODO: make this work for more than one assigned region
+            granularity = locations.get(restrictions[0], granularity)
+            # no need to filter data, as that's already done in load_data
+            chosen_region = restrictions[0]
+            
+            location_obj = Location.objects.get(name=chosen_region)
```

Comment:
Is it possible there could be more than one location with the same name? Would it be possible to key this off Location ID instead?

Reply:
Good point. It's definitely possible for location restrictions - changed the logic to pull in both name and ids. However, it's trickier for location-search and map-click based filtering, because those locations are all name based. Will need to give those some thought but problems for another day.